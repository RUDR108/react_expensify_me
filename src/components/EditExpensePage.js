import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import {editExpense,removeExpense} from '../actions/expenses';

const EditExpensePage=(props)=>{
    console.log(props);
return(
    <div>
        <button onClick={()=>{return  props.history.push('/'); }}>Click here to Select from HomePage.</button>
        <ExpenseForm expenses={props.expenses}
        submit={(expense)=>{
            props.dispatch(editExpense(props.expenses.id,expense));
            props.history.push('/');
        }}/>
        <button onClick={() => {
            props.dispatch(removeExpense({ id: props.expenses.id }));
            props.history.push('/');
          }}>Remove</button>
    </div>);
};

  
const mapStateToProps=(state,props)=>{
   return  {
        expenses:state.expenses.find((expense)=>{
            return expense.id===props.match.params.id
        })
    };
};

export default connect(mapStateToProps)(EditExpensePage);
