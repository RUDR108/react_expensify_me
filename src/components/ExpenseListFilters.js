import React from 'react';
import uuid from 'uuid';
import {DateRangePicker} from 'react-dates';
import { connect } from 'react-redux';
import {setTextFilter,sortByDate,sortByAmount} from '../actions/filters';
import {setStartDate,setEndDate} from '../actions/filters';

class ExpenseListFilter extends React.Component{
    state={
        calendarFocused:null
    };

    onDatesChange=({startDate,endDate})=>{
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };

    onFocusChange=(calendarFocused)=>{
        this.setState(()=>({calendarFocused}));
    }

render(){
    return(<div>
        <input type="text" 
        value={this.props.filters.sortBy} 
        onChange={(e)=>{this.props.dispatch(setTextFilter(e.target.value));}}/>
        <select onChange={(e)=>{
            if(e.target.value=='date'){
                this.props.diapatch(sortByDate());
            }else if(e.target.value=='amount'){
                this.props.dispatch(sortByAmount());
            }
        }}>
        <option value="date">Date</option>
        <option value="amount">Amount</option>
        </select>
        <DateRangePicker
        startDate={this.props.filters.startDate} // momentPropTypes.momentObj or null,
        startDateId={uuid()} // PropTypes.string.isRequired,
        endDate={this.props.filters.endDate} // momentPropTypes.momentObj or null,
        endDateId={uuid()}// PropTypes.string.isRequired,
        onDatesChange={this.onDatesChange} // PropTypes.func.isRequired,
        focusedInput={this.state.calendarFocused} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
        onFocusChange={this.onFocusChange} // PropTypes.func.isRequired,
        showClearDates={true}
        numberOfMonths={1}
        isOutsideRange={()=>false}
        />
    </div>);
    }
};

const mapStateToPraops=(state)=>{
    return{
        filters:state.filters
    };
}

export default connect(mapStateToPraops)(ExpenseListFilter);