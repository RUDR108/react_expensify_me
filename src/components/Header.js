import React from 'react';
import {NavLink} from  'react-router-dom';

const Header=()=>(
    <header>
    <h1>Expensify</h1>
    
    <NavLink to="/" activeClassName="is-active" exact={true}>GO home</NavLink><br/>
    <NavLink to="/create" activeClassName="is-active">Create_page</NavLink><br/>
    <NavLink to="/edit" activeClassName="is-active">Edit_Page</NavLink><br/>
    <NavLink to="/help" activeClassName="is-active">help</NavLink><br/>
    
    </header>
);

export default Header;