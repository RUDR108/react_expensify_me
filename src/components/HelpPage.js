import React from 'react';

const HelpPage=()=>(
    <div>This is from  Helppage.</div>
);

export default HelpPage;