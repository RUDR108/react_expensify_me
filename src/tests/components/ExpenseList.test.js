import React from 'react';
import { shallow } from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import {ExpenseListItem} from '../../components/ExpenseListItem'
  
const expenses=[{
    id:'123',
    description:'a description',
    note:'a note',
    amount:23,
    createdAt:0
},{
    id:'124',
    description:'a1 description',
    note:'a1 note',
    amount:34,
    createdAt:1000
},{
    id:'125',
    description:'a2 description',
    note:'a2 note',
    amount:21,
    createdAt:-1000
}]

test('should render ExpenseList with list',()=>{
    const wrapper=shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseList with empty message',()=>{
    const wrapper=shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});



