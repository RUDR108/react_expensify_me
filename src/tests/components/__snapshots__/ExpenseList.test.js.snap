// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `
<div>
  <Connect(ExpenseListItem)
    amount={23}
    createdAt={0}
    description="a description"
    id="123"
    key="123"
    note="a note"
  />
  <Connect(ExpenseListItem)
    amount={34}
    createdAt={1000}
    description="a1 description"
    id="124"
    key="124"
    note="a1 note"
  />
  <Connect(ExpenseListItem)
    amount={21}
    createdAt={-1000}
    description="a2 description"
    id="125"
    key="125"
    note="a2 note"
  />
</div>
`;

exports[`should render ExpenseList with list 1`] = `
<div>
  <Connect(ExpenseListItem)
    amount={23}
    createdAt={0}
    description="a description"
    id="123"
    key="123"
    note="a note"
  />
  <Connect(ExpenseListItem)
    amount={34}
    createdAt={1000}
    description="a1 description"
    id="124"
    key="124"
    note="a1 note"
  />
  <Connect(ExpenseListItem)
    amount={21}
    createdAt={-1000}
    description="a2 description"
    id="125"
    key="125"
    note="a2 note"
  />
</div>
`;
